Chirich/Lab1/Program.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using Crc32;
+
+namespace Lab1
+{
+    internal class Program
+    {
+        private static readonly Dictionary<string, HashAlgorithm> Algorithms = new Dictionary<string, HashAlgorithm>
+        {
+            {"sha1", new SHA1Managed()},
+            {"sha256", new SHA256Managed()},
+            {"md5", new SHA256Managed()},
+            {"crc32", new Crc32Algorithm()}
+        };
+
+        private static void Main(string[] args)
+        {
+            if (args.Length == 0 || args.Length == 1 && args[0] == "help")
+            {
+                Help();
+                return;
+            }
+
+            var filePath = args.Length >= 1 ? args[0] : string.Empty;
+            var blockSize = 1024;
+            if (args.Length >= 2) int.TryParse(args[1], out blockSize);
+            var algorithType = args.Length >= 3 ? args[2].ToLower() : "sha1";
+            var mode = args.Length >= 4 ? args[3].ToLower() : "sync";
```

Comment:
Actually only the last parameter should have default value, in other cases, for example, if you run the program with just filepath and blocksize (without specifying algorithm and mode) it should not try to guess other parameters.
Plus, default mode is async, according to task description.